include/engine/guidance/leg_assembler.hpp

```diff
@@ -0,0 +1,160 @@
+#ifndef ENGINE_GUIDANCE_LEG_ASSEMBLER_HPP_
+#define ENGINE_GUIDANCE_LEG_ASSEMBLER_HPP_
+
+#include "engine/guidance/route_leg.hpp"
+#include "engine/guidance/route_step.hpp"
+#include "engine/guidance/leg_geometry.hpp"
+#include "engine/internal_route_result.hpp"
+
+#include <boost/range/irange.hpp>
+
+#include <vector>
```

Comment:
cstdint for size_t, uint32_t
numeric for accumulate
algorithm for transform, sort, min
string for, well, string :)
iterator for next